### PR TITLE
Animate basic shapes

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18035)

### CODE REVIEW

**Comment by nox** [comment]
Cc @dadaa

**Comment by nox** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=9e42a940030271a5c4e5dc4a55a117f330ab9c26

File: components/style/values/generics/rect.rs [hunk 1]
```diff
@@ -51,6 +52,24 @@ impl<T> Rect<T>
...
+impl<L> Animatable for Rect<L>
+where
+    L: Animatable,
+{
+    fn add_weighted(
+        &self,
+        other: &Self,
+        self_portion: f64,
+        other_portion: f64,
+    ) -> Result<Self, ()> {
```
Comment (emilio) [inline components/style/values/generics/rect.rs]: `derive(Animatable)`? ;)

**Review by emilio — APPROVED** [review]
r=me, with a green gecko try.

**Comment by nox** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=22905db42de44a6fdfca4d543876d349b5bbe127

**Comment by nox** [comment]
Without `shape-outside` animations, this time.

https://treeherder.mozilla.org/#/jobs?repo=try&revision=e130f51512db2d9b126539d977b74239f394de1c

**Comment by hiikezoe** [comment]
You can just mark shape-outside as discrete type animation for now.

**Comment by nox** [comment]
> You can just mark shape-outside as discrete type animation for now.

I just kept `animation_value_type="none"`.

**Comment by nox** [comment]
Another last geckotry run, this time will all the missing `compute_distance` methods.

https://treeherder.mozilla.org/#/jobs?repo=try&revision=edc8169e789bb71d2e8b0a6558ff9c0d188ccc0d

**Comment by hiikezoe** [comment]
yeah, that's fine too, if you change it to 'discrete', some wpt can pass and test_transitions_per_property.html never fail.

**Comment by nox** [comment]
I set it to `discrete` and fixed the build error.

https://treeherder.mozilla.org/#/jobs?repo=try&revision=78e819b06044322fd4004f580a1d356844afd2af

**Comment by nox** [comment]
@bors-servo r=emilio

(Told me on IRC.)

**Comment by hiikezoe** [comment]
FWIW I did push expectation fix for web-platform-tests.

https://hg.mozilla.org/integration/autoland/rev/c73f64ef71d2d87f36008f5ff661f4050692bd06

**Pull request merged**